pkg/platform/kube/project/safecache.go

```diff
@@ -0,0 +1,97 @@
+package project
+
+import (
+	"sync"
+
+	"github.com/nuclio/nuclio/pkg/platform"
+)
+
+type SafeCache struct {
+	mu            sync.Mutex
+	projectsCache []platform.Project
+}
+
+func NewSafeCache() *SafeCache {
+	return &SafeCache{projectsCache: []platform.Project{}}
+}
+
+func (c *SafeCache) Add(projectInstance platform.Project) {
+	c.mu.Lock()
```

Comment:
use `defer c.mu.Unlock()` to avoid cases you receive a panic/error and the lock won't unlock